quinn/src/endpoint.rs

```diff
@@ -37,7 +38,7 @@ where
     S: proto::crypto::Session,
 {
     pub(crate) inner: EndpointRef<S>,
-    pub(crate) default_client_config: ClientConfig<S>,
+    pub(crate) default_client_config: OnceCell<ClientConfig<S>>,
```

Comment:
I'm not super sure about the global state here. Should it rather be a `Mutex<Option<ClientConfig>>`?

Reply:
What global state?

Reply:
Oh, I guess I misinterpreted it and thought `OnceCell`  would be like `lazy_static` and initialize things from a global reference.